firefox_puppeteer/api/places.py

```diff
@@ -89,13 +89,15 @@ def restore_default_bookmarks(self):
               let observer = {
                 observe: function (aSubject, aTopic, aData) {
                   Services.obs.removeObserver(observer, "bookmarks-restore-success");
+                  Services.obs.removeObserver(observer, "bookmarks-restore-failed");
 
-                  marionetteScriptFinished(true);
+                  marionetteScriptFinished(aTopic == "bookmarks-restore-success");
                 }
               };
 
               // Trigger the import of the default bookmarks
               Services.obs.addObserver(observer, "bookmarks-restore-success", false);
+              Services.obs.addObserver(observer, "bookmarks-restore-failed", false);
               BookmarkHTMLUtils.importFromURL(defaultBookmarks, true);
             """, script_timeout=10000)
         except TimeoutException:
```

Comment:
Now with the code changed above, we will have to get rid of the except block. If a timeout really happens we have to get it passed-through. Also in case of `execute_script` returns false, we have a raise an exception to indicate that the action we wanted to perform has been failed.